### ISSUE TITLE
new rule: late-return

### ISSUE THREAD

**Issue by chavacava** [mgechev/revive#404]
This new rule is inspired by [this question on twitter](https://twitter.com/francesc/status/1255620804705808384): 

> Is there a static analyzer for #golang that will warn about possible early returns?
```go
if foo {
 .... 
 return a
} else {
  return b
}
```
Should be:
```go
if ! foo {
  return b
}
...
return a
```
Contrary to other `revive` rules this is more a style-related rule. I'm not sure if it has its place in `revive`

**Comment by mgechev** [comment]
I think it makes sense to have it.
